components/autofill/src/sync/mod.rs

```diff
@@ -342,7 +342,7 @@ fn plan_incoming<T: std::fmt::Debug + SyncRecord>(
                                 mirror.as_ref().map(|m| m.metadata()),
                             );
                             IncomingAction::UpdateLocalGuid {
-                                old_guid,
```

Comment:
so in effect here, we were passing `old_guid: incoming_record.id().clone(), record: incoming_record` - and it would have been obvious why the assertion was failing. Hence my decision to remove that from the enum.